[google_maps_flutter] Improve README and API doc comments

filiph opened the pull request:
There are some inconsistencies and anachronisms in the various READMEs and in the API docs. This PR attempts to address them.

- Makes sure to link to platform-specific READMEs from the main README file.
- Makes the READMEs more consistent (e.g. only leaving the bare minimum quick start for the main README, and leaving all platform-specific notes to platform READMEs).
- Using the more correct “Not supported on all platforms” instead of “Supported on Android only” for functionality that works on both Android and the Web. Intentionally staying vague in API docs in order to make later maintenance easier. (https://github.com/flutter/flutter/issues/161948#issuecomment-2605429688)
- Remove a doc comment from before null safety about asserting non-null.
- Format the platform sections in the main README so that they’re proper lists, and so that all of them end with a link to the appropriate platform-specific README.

Issues:

- Fixes flutter/flutter#161954
- Related: flutter/flutter#161950, flutter/flutter#161948

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

filiph commented:
I see that an automated CHANGELOG check is failing. It felt weird to be updating the otherwise-laconic CHANGELOGs with documentation changes but I'm happy to do that.

stuartmorgan commented:
> I see that an automated CHANGELOG check is failing.

See https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#faq

stuartmorgan commented on packages/google_maps_flutter/google_maps_flutter/README.md:
```diff
@@ -32,84 +41,82 @@ For more details, see [Getting started with Google Maps Platform](https://develo
 
 1. Set the `minSdkVersion` in `android/app/build.gradle`:
```
You can just remove this whole bullet point while you are touching this; Flutter itself only supports 21+ now.

filiph replied:
Done.

stuartmorgan commented on packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart:
```diff
@@ -261,6 +264,8 @@ class GoogleMap extends StatefulWidget {
   /// * The My Location button animates to focus on the user's current location
   /// if the user's location is currently known.
   ///
+  /// This feature is not present in the Google Maps SDK for the web.
```
Ideally we wouldn't reference a specific platform here, but the ship has sailed on that below so we can do this now, and revisit the whole thing in the future.

stuartmorgan commented on packages/google_maps_flutter/google_maps_flutter_android/README.md:
```diff
@@ -78,6 +78,10 @@ Google Play the latest renderer will not be available and the legacy renderer wi
 WARNING: `AndroidMapRenderer.legacy` is known to crash apps and is no longer supported by the Google Maps team
 and therefore cannot be supported by the Flutter team.
 
+#### Cloud-based map styling
```
The previous heading is `##`, so this should be `###`.

filiph replied:
Done.

stuartmorgan commented on packages/google_maps_flutter/google_maps_flutter_android/README.md:
```diff
@@ -78,6 +78,10 @@ Google Play the latest renderer will not be available and the legacy renderer wi
 
+#### Cloud-based map styling
+
+Cloud-based map styling works on Android only if `AndroidMapRenderer.latest` map renderer has been initialized.
```
I think people may read this as having to explicitly initialize the `latest` renderer, but in most cases it's the default now. How about "Cloud-based map styling is not supported with the `legacy` renderer."

filiph replied:
Done.

filiph commented:
Addressed feedback and incremented versions using `update-release-info`. The CI/CD for changelog is now green but it's failing in Firebase Test Lab. Snip:

```
More details are available at [ https://console.firebase.google.com/project/flutter-infra-staging/testlab/histories/bh.8eb89ae93eb3eb36/matrices/7103368191579169772 ].
┌─────────┬────────────────────────┬─────────────────────┐
│ OUTCOME │    TEST_AXIS_VALUE     │     TEST_DETAILS    │
├─────────┼────────────────────────┼─────────────────────┤
│ Failed  │ panther-33-en-portrait │ Application crashed │
└─────────┴────────────────────────┴─────────────────────┘
```

I don't have access to https://console.firebase.google.com/project/flutter-infra-staging/testlab/histories/bh.8eb89ae93eb3eb36/matrices/7103368191579169772 to check what exactly went wrong. Did I fudge something? I believe I only changed documentation but I'm a bit nervous because the failure is at `packages/google_maps_flutter/google_maps_flutter_android` which is one of the packages I touched.

reidbaker commented:
Logcat and video from the test attached. 
The "overview" shows `java.lang.IllegalStateException: Trying to call onSurfaceCreated with no current context`

https://github.com/user-attachments/assets/03f0c272-05f2-4e51-b58e-51968269c92c

[plugins_android_test_packages_google_maps_flutter_google_maps_flutter_android_unknown_build_3e8e411b-bfa1-4d48-8d13-793bec41437e_example_0_panther-33-en-portrait_logcat.txt](https://github.com/user-attachments/files/18735706/plugins_android_test_packages_google_maps_flutter_google_maps_flutter_android_unknown_build_3e8e411b-bfa1-4d48-8d13-793bec41437e_example_0_panther-33-en-portrait_logcat.txt)


Have not dug in but the failure looks related.

filiph commented:
I just went through the CL line by line and, FWIW, the only changes are:

* README + CHANGELOG files
* doc comments in .dart files
* version bump

stuartmorgan reviewed (approved):
LGTM

jokerttu commented on packages/google_maps_flutter/google_maps_flutter/README.md:
```diff
@@ -30,86 +39,72 @@ For more details, see [Getting started with Google Maps Platform](https://develo
-Check [the `google_maps_flutter_web` README](https://pub.dev/packages/google_maps_flutter_web)
-for the latest information on how to prepare your App to use Google Maps on the
-web.
+2. Read about iOS-specific features and limitations in the
```
@filiph
I know this PR is merged already, but if you still work with documentation this could be fixed on next PR.
This should be `web` specific in this context.

Also [changelog entry ](https://github.com/flutter/packages/pull/8560/files#diff-125e4c4c56d466e14364386e43fc727a35e14f9874bb14407d8703e26fe7ce8aR8)was for old version was changed on this PR as well.

filiph replied:
Good catch, thanks! Opened https://github.com/flutter/packages/pull/8754.
